src/edge_detection.h

```diff
@@ -50,31 +51,39 @@ struct EdgeParameter
     void verify() const; // self-verification that all parameters are correct
 };
 
+// Optional parameter to apply before looking for first and second gradients on an image (findEdgePoints)
+enum Filter
+{
+    NONE = 0,
+    MEDIAN,
+    GAUSSIAN
+};
+
 template <typename _Type>
 class EdgeDetection;
 
 class EdgeDetectionHelper
 {
 public:
     static void find( EdgeDetection<double> & edgeDetection, const PenguinV_Image::Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height,
-                      const EdgeParameter & edgeParameter );
+                      const EdgeParameter & edgeParameter, Filter filterToApply, uint32_t filterKernelSize, double sigma );
 
     static void find( EdgeDetection<float> & edgeDetection, const PenguinV_Image::Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height,
-                      const EdgeParameter & edgeParameter );
+                      const EdgeParameter & edgeParameter, Filter filterToApply, uint32_t filterKernelSize, float sigma );
 };
 
 template <typename _Type = double>
 class EdgeDetection
 {
 public:
-    void find( const PenguinV_Image::Image & image, const EdgeParameter & edgeParameter = EdgeParameter() )
+    void find( const PenguinV_Image::Image & image, const EdgeParameter & edgeParameter = EdgeParameter(), Filter filterToApply = NONE, uint32_t filterKernelSize = 3u, _Type sigma = 1.0f )
     {
-        find( image, 0, 0, image.width(), image.height(), edgeParameter );
+        find( image, 0, 0, image.width(), image.height(), edgeParameter, filterToApply, filterKernelSize, sigma );
     }
 
-    void find( const PenguinV_Image::Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height, const EdgeParameter & edgeParameter = EdgeParameter() )
+    void find( const PenguinV_Image::Image & image, uint32_t x, uint32_t y, uint32_t width, uint32_t height, const EdgeParameter & edgeParameter = EdgeParameter(), Filter filterToApply = NONE, uint32_t filterKernelSize = 3u, _Type sigma = 1.0f )
     {
-        EdgeDetectionHelper::find( *this, image, x, y, width, height, edgeParameter );
+        EdgeDetectionHelper::find( *this, image, x, y, width, height, edgeParameter, filterToApply, filterKernelSize, sigma );

```

Comment:
What if we clear our 2 vector here just before calling EdgeDetectionHelper? I think it would make code cleaner as well.